src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts

```diff
@@ -0,0 +1,74 @@
+import { Token } from '@uniswap/sdk-core'
+import { atom, useAtom } from 'jotai'
+import { loadable } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+const BASE_URL = 'https://temp.api.uniswap.org/v1/tokens/search?tokenQuery='
+
+type Address = `0x${string}`
+
+interface FetchTokensResult {
+  chainId: number
+  name: string
+  address: Address
+  decimals: number
+  symbol: string
+  logoURI: string
+  coinGeckoId: string
+  priceUsd: number
+  price24hChange: number
+  volume24h: number
+  marketCap: number
+}
+
+interface FetchTokensApiResult {
+  code: number
+  status: 'success' | 'failed'
+  hasNext: boolean
+  data: FetchTokensResult[]
+}
+
+async function fetchTokens(query: string): Promise<FetchTokensApiResult> {
+  const response = await fetch(`${BASE_URL}${query}`)
+  const data = await response.json()
+
+  return data
+}
+
+const searchQuery = atom('')
+
+const tokensData = loadable(
```

Comment:
Async atom, nice!

Reply:
>  So, I change the request to be in a follow up PR rather than here :)
Great point, happy to follow up on this. I see quite a bit of refactor potential here as well; so can follow up with that (and should help reduce our dependency on the uniswap's base)